### PR TITLE
Add release automation using JReleaser

### PR DESCRIPTION
**Pull request by wendigo**
This adds a release automation to the airlift repository. Only maintainers will be able to schedule a release.

The required secrets are:

1. JRELEASER_GPG_PUBLIC_KEY: armored public GPG key
2. JRELEASER_GPG_SECRET_KEY: armored private GPG key
3. JRELEASER_GPG_PASSPHRASE: private GPG passphrase
4. JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME: OSS sonatype username
5. JRELEASER_NEXUS2_MAVEN_CENTRAL_TOKEN: OSS sonatype token

### CODE REVIEW

File: jreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      failOnWarning: false # We don't want to fail the build on warnings
+      failOnError: false # We don't want to fail the build on errors
+    nexus2:
+      maven-central:
+        active: ALWAYS
+        url: https://oss.sonatype.org/service/local
+        snapshotUrl: https://oss.sonatype.org/content/repositories/snapshots/
+        closeRepository: true
+        javadocJar: false # Not every module has javadoc (packaging)
+        releaseRepository: false # Switch to true to release to Central
```
Comment (wendigo) [inline jreleaser.yml:59]: This is now configure not to release a staging repository. This is done by hand in the OSS UI

**Comment by wendigo** [comment]
Successful run on my fork: https://github.com/wendigo/airlift/actions/runs/10170260161

**Comment by wendigo** [comment]
Created a Github release: https://github.com/wendigo/airlift/releases (empty because there were no PRs merged between last and released tags)

**Comment by wendigo** [comment]
Staged repository with all the artefacts (I've skipped some when relocating to different group id): https://oss.sonatype.org/content/repositories/plwendigo-1247/

**Comment by wendigo** [comment]
Screenshot 2024-07-31 at 00 02 14

**Comment by wendigo** [comment]
Screenshot 2024-07-31 at 00 03 24

**Comment by wendigo** [comment]
@electrum @dain please approve so I can start doing tests :)

**Review by electrum — APPROVED** [review]

**Pull request merged**